examples/file_like.py

```diff
@@ -0,0 +1,245 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+===================================================================
+Streaming data through file-like support
+===================================================================
+
+In this example, we will show how to decode streaming data. That is, when files
+do not reside locally, we will show how to only download the data segments that
+are needed to decode the frames you care about. We accomplish this capability
+with Python
+`file-like objects <https://docs.python.org/3/glossary.html#term-file-like-object>`_."""
```

Comment:
Maybe indicate that we'll be using the `VideoDecoder` in this example, but everything also works for `AudioDecoder`?

Also: we should update this link in the `VideoDecoder` docstring, and create a similar one for `AudioDecoder`:

https://github.com/pytorch/torchcodec/blob/2c137e75661fd3c04b38901480defdf86a9183a9/src/torchcodec/decoders/_video_decoder.py#L32

Reply:
Keeping unresolved until I actually replace the text in the source files.